File: Python/DeleteColumnsToMakeSorted.py
# https://leetcode.com/problems/delete-columns-to-make-sorted/description/?envType=daily-question&envId=2025-12-20

class Solution:
    def minDeletionSize(self, strs: List[str]) -> int:
        res = 0

        for col in range(len(strs[0])):
            for row in range(1, len(strs)):
                if(strs[row][col] < strs[row - 1][col]):
                    res += 1
                    break
        
        return res
    
if __name__ == "__main__":
    from typing import List
    size = int(input())
    strs = []
    for _ in range(size):
        strs.append(input().strip())
    sol = Solution()
    ans = sol.minDeletionSize(strs)
    print(ans)

File: DeleteColumnsToMakeSorted.java
// https://leetcode.com/problems/delete-columns-to-make-sorted/description/?envType=daily-question&envId=2025-12-20

public class DeleteColumnsToMakeSorted {
    public int minDeletionSize(String[] strs) {
        int res = 0;
        for(int j = 0; j < strs[0].length(); j++){
            res = res + isUnsorted(strs, j);
        }
        return res;
    }

    private int isUnsorted(String[] strs, int j){
        for(int i = 1; i < strs.length; i++){
            if(strs[i].charAt(j) < strs[i - 1].charAt(j)){
                return 1;
            }
        }
        return 0;
    }

    public static void main(String[] args){
        DeleteColumnsToMakeSorted solution = new DeleteColumnsToMakeSorted();
        String[] strs = {"cba", "daf", "ghi"};
        int result = solution.minDeletionSize(strs);
        System.out.println("Minimum number of columns to delete: " + result);
    }
}
